Neos.ContentRepository.Core/Classes/ContentRepository.php

```diff
@@ -89,64 +91,77 @@ public function __construct(
      */
     public function handle(CommandInterface $command): void
     {
-        $command = $this->commandHook->onBeforeHandle($command);
-        $privilege = $this->authProvider->canExecuteCommand($command);
-        if (!$privilege->granted) {
-            throw AccessDenied::becauseCommandIsNotGranted($command, $privilege->getReason());
-        }
-
-        $toPublish = $this->commandBus->handle($command);
-        $correlationId = CorrelationId::fromString(sprintf('%s_%s', substr($command::class, strrpos($command::class, '\\') + 1, 20), bin2hex(random_bytes(9))));
-
-        // simple case
-        if ($toPublish instanceof EventsToPublish) {
-            $this->eventStore->commit($toPublish->streamName, $this->enrichAndNormalizeEvents($toPublish->events, $correlationId), $toPublish->expectedVersion);
-            $fullCatchUpResult = $this->subscriptionEngine->catchUpActive(); // NOTE: we don't batch here, to ensure the catchup is run completely and any errors don't stop it.
-            if ($fullCatchUpResult->hadErrors()) {
-                throw CatchUpHadErrors::createFromErrors($fullCatchUpResult->errors);
+        $this->tracer->span('ContentRepository::handle', ['c' => get_class($command)], function () use ($command) {
```

Comment:
Instead of a `NullTracer`, `$this->tracer?->span(…)` might be faster, as it means less instructions?